src/core/annotation.js

```diff
@@ -955,6 +958,8 @@ var PopupAnnotation = (function PopupAnnotationClosure() {
       return;
     }
 
+    var parentSubtype = parentItem.get('Subtype');
+    this.data.parentType = isName(parentSubtype) ? parentSubtype.name : null;
     this.data.parentId = dict.getRaw('Parent').toString();
```

Comment:
*This is __completely out-of-scope for this PR__, but it's something that I noticed which we might want try and address in a future PR.*

Given that there are bad PDF files where annotations are not `Ref`s, see e.g. PR #7570, this probably doesn't work very well in that (rare) case. Ideally I think that we probably want to somehow tap into the `id` as generated above, but it might not be entirely simple given that annotations can be ordered arbitrarily.
Anyway, probably worth looking into later...